perfkitbenchmarker/analyze.py

```diff
@@ -0,0 +1,202 @@
+# Copyright 2016 PerfKitBenchmarker Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Functions for analyzing output from benchmarks.
+
+In general, this module works on data in the form of Pandas data
+frames, because that seems to be the most standard Python statistical
+data format.
+"""
+
+import pandas as pd
+
+from perfkitbenchmarker import sample
+
+
+def AllStreamsInterval(start_times, durations, stream_ids):
+  """Compute when all streams were active from multistream records.
+
+  Args:
+    start_times: a pd.Series of POSIX timestamps, as floats.
+    durations: a pd.Series of durations, as floats measured in seconds.
+    stream_ids: a pd.Series of any type.
+
+  All three arguments must have matching indices. Each (start_time,
+  duration, stream_id) triple is considered a record of a single
+  operation performed by the given stream. This function computes the
+  interval from when the last stream to start began its first
+  operation to when the first stream to stop finished its last
+  operation.
+
+  Returns: a tuple of
+    - a float holding the POSIX timestamp when the last stream to
+      start began its first operation
+    - a float holding the length of time in seconds from the first
+      return value until the first proceess to end stopped its last
+      operation
+  """
+
+  stream_start_times = start_times.groupby(stream_ids).min()
+  stream_end_times = (start_times + durations).groupby(stream_ids).max()
+  interval_start = stream_start_times.max()
+  interval_end = stream_end_times.min()
+
+  return interval_start, interval_end - interval_start
+
+
+def StreamStartAndEndGaps(start_times, durations,
```

Comment:
I had them together originally, and split them because the combined function had four return values and did several different things, and I thought it was hard to understand what it did or how to use it. I think it's more confusing than it's worth, but if you really want, I will merge them.